client/state/data-layer/wpcom/plans/index.js

```diff
@@ -1,32 +1,62 @@
 /**
  * Internal dependencies
  */
-import wpcom from 'lib/wp';
+import { fetch } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { PLANS_REQUEST } from 'state/action-types';
 import {
-	PLANS_RECEIVE,
-	PLANS_REQUEST,
-	PLANS_REQUEST_SUCCESS,
-	PLANS_REQUEST_FAILURE
-} from 'state/action-types';
+	plansReceiveAction,
+	plansRequestFailureAction,
+	plansRequestSuccessAction,
+} from 'state/plans/actions';
 
-export const requestPlans = ( { dispatch } ) => (
-	wpcom
-		.withLocale()
-		.plans()
-		.list( { apiVersion: '1.4' } )
-		.then( plans => {
-			dispatch( { type: PLANS_REQUEST_SUCCESS } );
-			dispatch( { type: PLANS_RECEIVE, plans } );
-		} )
-		.catch( rawError => {
-			const error = rawError instanceof Error
-				? rawError.message
-				: rawError;
+/**
+ * Dispatches a request to fetch all available WordPress.com plans
+ *
+ * @param {Function} dispatch Redux dispatcher
+ * @param {Object} action Redux action
+ * @returns {Object} HTTP request action
+ */
+export const requestPlans = ( { dispatch }, action ) =>
+	dispatch( fetch( {
+		apiVersion: '1.4',
+		method: 'GET',
+		path: '/plans',
```

Comment:
To be honest, almost everything except this line and the `receivePlans → plansReceiveAction` in this file seems like incidental complexity – can we have some sensible defaults, so that we don’t need almost any of it?